src/gallium/state_trackers/nine/swapchain9.c

```diff
@@ -132,6 +131,12 @@ NineSwapChain9_Resize( struct NineSwapChain9 *This,
                 (pParams->SwapEffect == D3DSWAPEFFECT_COPY) ||
                 (pParams->SwapEffect == D3DSWAPEFFECT_DISCARD), D3DERR_INVALIDCALL);
 
+    if (pDevice->ex && pParams->BackBufferCount >= SWAPCHAIN_BUFFERS_MAX) {
```

Comment:
Why not just use these:
#define D3DPRESENT_BACK_BUFFERS_MAX    3
#define D3DPRESENT_BACK_BUFFERS_MAX_EX 30

If not ex return error when using more than D3DPRESENT_BACK_BUFFERS_MAX, and return error if using ex and more than D3DPRESENT_BACK_BUFFERS_MAX_EX buffers.
Use D3DPRESENT_BACK_BUFFERS_MAX_EX+1 slots for the table size.